### PR TITLE
[INLONG-9225][Audit] Automatically create audit topic after service startup

### LINKED ISSUE #9225: [Improve][Audit] Automatically create audit topic after service startup
### Description

Currently, for audit services, topics need to be manually created. If they are not manually created or if the user is unaware of the need for manual creation, statistical information cannot be generated, and the corresponding tables have no data, which affects functionality. This modification is to use the program to create a topic and avoid this issue.

table:
df9c686dee2fca1ffb207df2505f2f0

warn log：
8ddbb0178f31e6964b8ce69cabf43b7

After resolving the above issues：
image
image
image

### InLong Component

InLong Audit

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by LiJie20190102**
Automatically create audit topic after service startup

### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-9225] [inlong-audit]Automatically create audit topic after service startup

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #9225 

### Motivation

Currently, for audit services, topics need to be manually created. If they are not manually created or if the user is unaware of the need for manual creation, statistical information cannot be generated, and the corresponding tables have no data, which affects functionality. This modification is to use the program to create a topic and avoid this issue.

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ √] This change is already covered by existing tests, such as:
  *(please describe tests)*
1. Modify audit. kafka. topic in the application. properties file, such as audit. kafka. topic=audit_ Topic_ Test (provided that the topic is not in kafka);

2. Restart the audit service and check if the topic was successfully created

image
image

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-audit/audit-store/src/main/java/org/apache/inlong/audit/service/consume/KafkaConsume.java [hunk 1]
```diff
@@ -67,16 +77,70 @@ public void start() {
...
+
+            NewTopic needCreateTopic = new NewTopic(topic, partition, (short) factor);
+
+            CreateTopicsResult createTopicsResult =
+                    adminClient.createTopics(Collections.singletonList(needCreateTopic));
+            createTopicsResult.all().get();
+
+        } catch (ExecutionException | InterruptedException e) {
+            throw new RuntimeException(
+                    String.format("create audit topic:{} error with config:%s", getProperties(mqConfig)));
```
Comment (healchow) [inline inlong-audit/audit-store/src/main/java/org/apache/inlong/audit/service/consume/KafkaConsume.java:132]: The `{}` seems not used in here, and suggested print the stacktrace into the log.

File: inlong-audit/audit-store/src/main/java/org/apache/inlong/audit/service/consume/KafkaConsume.java [hunk 2]
```diff
@@ -67,16 +77,70 @@ public void start() {
...
+
+            if (topicSet.contains(topic)) {
+                // not need
+                LOG.info("The audit topic:{} already exists.", topic);
+                return;
+            }
+
+            DescribeClusterResult describeClusterResult = adminClient.describeCluster();
+            Collection<Node> nodes = describeClusterResult.nodes().get();
+            if (nodes.isEmpty())
```
Comment (healchow) [inline inlong-audit/audit-store/src/main/java/org/apache/inlong/audit/service/consume/KafkaConsume.java:118]: Please add `{}` for the `if` statement.

File: inlong-audit/audit-store/src/main/java/org/apache/inlong/audit/service/consume/KafkaConsume.java [hunk 3]
```diff
@@ -67,16 +77,70 @@ public void start() {
...
                 "KafkaConsume_Fetcher_Thread");
         thread.start();
     }
 
+    /**
+     * create topic if need
+     */
+    private void createTopic() {
+        int defaultNumPartitions = 3;
+        int defaultReplicationFactor = 2;
```
Comment (healchow) [inline inlong-audit/audit-store/src/main/java/org/apache/inlong/audit/service/consume/KafkaConsume.java:95]: I suggest extracting those two parameters as global constants in this class.

**Review by healchow — CHANGES_REQUESTED** [review]

**Comment by LiJie20190102** [comment]
@healchow ths , please review again

**Review by healchow — DISMISSED** [review]
LGTM

**Review by haifxu — DISMISSED** [review]

**Review by doleyzi — APPROVED** [review]

**Comment by dockerzhang** [comment]
please add the new configurations in the `application.properties`.

**Comment by LiJie20190102** [comment]
> please add the new configurations in the `application.properties`.

OK

**Review by haifxu — APPROVED** [review]

**Review by dockerzhang — DISMISSED** [review]

File: inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/KafkaSink.java [hunk 4]
```diff
@@ -282,6 +310,43 @@ private void initTopicProducer(String topic) {
...
+            }
+
+            DescribeClusterResult describeClusterResult = adminClient.describeCluster();
+            Collection<Node> nodes = describeClusterResult.nodes().get();
+            if (nodes.isEmpty()) {
+                throw new IllegalArgumentException("kafka server not find");
+            }
+
+            int partition = Math.min(topicPartitions, nodes.size());
+            int factor = nodes.size() > 1 ? Math.min(topicReplications, nodes.size()) : 1;
```
Comment (aloyszhang) [inline inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/KafkaSink.java:336]: `int factor = Math.min(topicReplications, nodes.size());` is enough？

Comment (LiJie20190102) [reply inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/KafkaSink.java]: ths

**Review by haifxu — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**